### PR TITLE
feat(UI-973): re-write sidebar to use popover

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
Re-write the sidebar to use popover to have better:
* On hover behavior with more convenient "enter-polygon"

## Linear Ticket
https://linear.app/autokitteh/issue/UI-973/rewrite-sidebar-to-use-popover

## What type of PR is this? (check all applicable)

-   [x] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/molecules/menu/menu.tsx [hunk 1]
```diff
@@ -88,39 +67,47 @@ export const Menu = ({ className, isOpen = false, onMouseLeave, onSubmenu }: Men
...
-								>
-									{t("myProjects")}
-								</motion.span>
-							) : null}
-						</AnimatePresence>
-					</Button>
-				</li>
+				<PopoverList animation="slideFromLeft">
+					<PopoverListTrigger>
+						<li className="group static">
```
Comment (J1za) [inline src/components/molecules/menu/menu.tsx:72]:
```suggestion
						<li className="group">
```

File: src/components/molecules/menu/menu.tsx [hunk 2]
```diff
@@ -88,39 +67,47 @@ export const Menu = ({ className, isOpen = false, onMouseLeave, onSubmenu }: Men
...
+							<div className="cursor-pointer before:absolute before:left-0 before:h-10 before:w-full" />
```
Comment (J1za) [inline src/components/molecules/menu/menu.tsx:73]: what this?

File: src/components/molecules/menu/menu.tsx [hunk 3]
```diff
@@ -88,39 +67,47 @@ export const Menu = ({ className, isOpen = false, onMouseLeave, onSubmenu }: Men
...
+					</PopoverListTrigger>
+					<PopoverListContent
+						className="z-30 flex h-screen flex-col rounded-lg bg-white px-4 pt-[212px] text-black"
+						itemClassName={cn(
+							"flex cursor-pointer items-center gap-2.5 rounded-3xl p-2 transition",
+							"hover:text-current text-center text-gray-1100 duration-300 hover:bg-gray-1250",
+							"text-fira-code whitespace-nowrap px-4 text-gray-1100 hover:bg-green-200 max-w-245 overflow-hidden"
+						)}
+						items={sortedProjectsList.map(({ id, name }) => ({ id, label: name, value: id }))}
+						onItemSelect={({ id: projectId }) => navigate(`/${SidebarHrefMenu.projects}/${projectId}/code`)}
```
Comment (J1za) [inline src/components/molecules/menu/menu.tsx:108]: if this list will be used only in the sidebar. This prop doesn't need

File: src/components/molecules/popover/popoverListContent.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+	};
+
+	const onKeyDown = (event: React.KeyboardEvent, item: PopoverListItem, index: number) => {
+		if (event.key === "Enter" || event.key === " ") {
+			handleItemClick(item, index, item.id);
+		} else if (event.key === "ArrowDown") {
+			context.setActiveIndex(Math.min(index + 1, items.length - 1));
+		} else if (event.key === "ArrowUp") {
+			context.setActiveIndex(Math.max(index - 1, 0));
+		}
```
Comment (J1za) [inline src/components/molecules/popover/popoverListContent.tsx:44]:
```suggestion
		if (event.key === "Enter" || event.key === " ") {
    handleItemClick(item, index, item.id);
} 
if (event.key === "ArrowDown") {
    context.setActiveIndex(Math.min(index + 1, items.length - 1));
} 
if (event.key === "ArrowUp") {
    context.setActiveIndex(Math.max(index - 1, 0));
}
```

File: src/components/molecules/popover/popoverListContent.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+					})}
+					key={item.id}
+					onClick={() => handleItemClick(item, index, item.id)}
+					onKeyDown={(event) => onKeyDown(event, item, index)}
+					ref={(node) => {
+						if (node) listRef.current[index] = node;
+					}}
+					role="option"
+					style={{
+						cursor: "pointer",
```
Comment (J1za) [inline src/components/molecules/popover/popoverListContent.tsx:63]: add to className

File: src/hooks/usePopover.ts [hunk 6]
```diff
@@ -30,36 +31,108 @@ export function usePopover({ initialOpen = false, modal, placement = "bottom" }:
...
+				initial: { opacity: 0, top: "50px" },
+				close: { opacity: 0, top: "50px" },
+			};
+			break;
+		case "none":
+		default:
+			break;
+	}
+
+	const { isMounted, styles } = useTransitionStyles(context, transitionConfiguration);
```
Comment (J1za) [inline src/hooks/usePopover.ts:63]:
```suggestion
	const animationConfigurations = {
  slideFromLeft: {
    open: { opacity: 1, left: 0 },
    initial: { opacity: 0, left: "-212px" },
    close: { opacity: 0, left: "-212px" },
  },
  slideFromBottom: {
    open: { opacity: 1, top: 0 },
    initial: { opacity: 0, top: "50px" },
    close: { opacity: 0, top: "50px" },
  },
};

const transitionConfiguration = animationConfigurations[animation] || {};

const { isMounted, styles } = useTransitionStyles(context, transitionConfiguration);
```

File: src/interfaces/components/popover.interface.ts [hunk 7]
```diff
@@ -6,9 +6,22 @@ export interface PopoverOptions {
 	modal?: boolean;
 	open?: boolean;
 	onOpenChange?: (open: boolean) => void;
+	animation?: "none" | "slideFromLeft" | "slideFromBottom";
```
Comment (J1za) [inline src/interfaces/components/popover.interface.ts:9]: I think "none" does't need

**Comment by J1za** [comment]
twitches when opening and closing 
chrome-capture-2024-12-2

File: src/components/molecules/menu/menu.tsx (hunk 1 above)
Comment (RonenMars) [reply src/components/molecules/menu/menu.tsx:72]: removed

File: src/components/molecules/menu/menu.tsx (hunk 2 above)
Comment (RonenMars) [reply src/components/molecules/menu/menu.tsx:73]: removed

File: src/components/molecules/menu/menu.tsx [hunk 8]
```diff
@@ -88,39 +67,47 @@ export const Menu = ({ className, isOpen = false, onMouseLeave, onSubmenu }: Men
...
+											{t("myProjects")}
+										</motion.span>
+									) : null}
+								</AnimatePresence>
+							</div>
+						</li>
+					</PopoverListTrigger>
+					<PopoverListContent
+						className="z-30 flex h-screen flex-col rounded-lg bg-white px-4 pt-[212px] text-black"
+						itemClassName={cn(
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/molecules/menu/menu.tsx:102]: Fixed

File: src/components/molecules/menu/menu.tsx (hunk 3 above)
Comment (RonenMars) [reply src/components/molecules/menu/menu.tsx:108]: What do you mean? this is practically our `onClick`

File: src/components/molecules/popover/popoverContentBase.tsx [hunk 9]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+
+import { FloatingFocusManager, FloatingPortal } from "@floating-ui/react";
+
+import { useMergeRefsCustom } from "@components/molecules/popover/utilities/mergeRefs";
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/molecules/popover/popoverContentBase.tsx:5]: done

File: src/components/molecules/popover/popoverListContent.tsx (hunk 4 above)
Comment (RonenMars) [reply src/components/molecules/popover/popoverListContent.tsx:44]: done

File: src/components/molecules/popover/popoverListContent.tsx (hunk 5 above)
Comment (RonenMars) [reply src/components/molecules/popover/popoverListContent.tsx:63]: Done

File: src/hooks/usePopover.ts (hunk 6 above)
Comment (RonenMars) [reply src/hooks/usePopover.ts:63]: done

File: src/interfaces/components/popover.interface.ts (hunk 7 above)
Comment (RonenMars) [reply src/interfaces/components/popover.interface.ts:9]: removed

File: src/components/molecules/menu/menu.tsx (hunk 3 above)
Comment (J1za) [reply src/components/molecules/menu/menu.tsx]: all right

File: src/components/molecules/menu/menu.tsx [hunk 10]
```diff
@@ -108,39 +87,48 @@ export const Menu = ({ className, isOpen = false, onMouseLeave, onSubmenu }: Men
...
+										>
+											{t("myProjects")}
+										</motion.span>
+									) : null}
+								</AnimatePresence>
+							</div>
+						</li>
+					</PopoverListTrigger>
+					<PopoverListContent
+						className="z-30 flex h-screen flex-col overflow-scroll rounded-lg bg-white px-4 pb-16 pt-[212px] text-black"
```
Comment (J1za) [inline src/components/molecules/menu/menu.tsx:120]:
```suggestion
						className="scrollbar z-30 flex h-screen flex-col overflow-scroll rounded-lg bg-white px-4 pb-16 pt-[212px] text-black"
```

Comment (RonenMars) [reply src/components/molecules/menu/menu.tsx:120]: [done](url)

**Review by J1za — APPROVED** [review]

**Pull request merged**